libraries/mbed/targets/hal/TARGET_Freescale/TARGET_KLXX/TARGET_KL25Z/serial_api.c

```diff
@@ -78,7 +79,17 @@ void serial_init(serial_t *obj, PinName tx, PinName rx) {
     }
     // Disable UART before changing registers
     obj->uart->C2 &= ~(UARTLP_C2_RE_MASK | UARTLP_C2_TE_MASK);
-    
+
+    // Enable UART transmitter to ensure TX activity is finished
+    obj->uart->C2 |= UARTLP_C2_TE_MASK;
+
+    // Wait for TX activity to finish
+    while(!(obj->uart->S1 & UARTLP_S1_TC_MASK));
+
+    // Disbale UARTs again
+    obj->uart->C2 &= ~(UARTLP_C2_RE_MASK | UARTLP_C2_TE_MASK);
+
```

Comment:
First you disable tx/rx to put it in a known state (though this of all lines could be omitted).
Then you enable TX. (In hind sight these two commands could probably just be reduced to disable RX enable TX).

Then you wait for TX activity to finish. This is important for the following case: https://github.com/mbedmicro/mbed/pull/1175/files#diff-f6f99e596c130732dd21269ad5d238f5R21

If you don't wait for the TX activity to complete before disabling TX, the previous message ("TX OK - Expected\r\n") won't be sent properly. This ensures that the last message is sent.

Once the TX activity is finished, you disable TX and RX again to put it in a known state.